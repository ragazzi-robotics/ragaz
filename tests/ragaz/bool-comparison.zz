def main():

    # equal, int/bool
    print(1 == 3)
    print(2 == 2)
    print(True == True)
    print(False == True)

    # equal, str
    print("a" == "b")
    print("aa" == "b")
    print("aa" == "bb")
    print("a" == "a")

    # not_equal
    print(1 != 4)
    print(False != False)
    print("a" != "b")

    # lower_than, float and int
    print(1 < 4)
    print(2 < 2.0)
    print(5.5 < 4)

    # lower_equal, float and int
    print(2 <= 2.0)
    print(3 <= 4)
    print(6.0 <= 4)

    # lower_than, str
    print("a" < "aa")
    print("a" < "b")
    print("a" < "a")
    print("b" < "a")

    # lower_equal, str
    print("a" <= "a")

    # greater_than, int
    print(1 > 4)
    print(2 > 2)
    print(5 > 4)

    # greater_equal, int
    print(2 >= 2)
    print(3 >= 4)
    print(6 >= 4)

    # greater_than, str
    print("a" > "aa")
    print("a" > "b")
    print("a" > "a")
    print("b" > "a")

    # lower_equal, str
    print("a" >= "a")
