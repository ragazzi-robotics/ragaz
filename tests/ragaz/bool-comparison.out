False
True
True
False
False
False
False
True
True
False
True
True
False
False
True
True
False
True
True
False
False
True
False
False
True
True
False
True
False
False
False
True
True
